### DISCUSSION TITLE
Passing UserData from Lua without cloning?

### DISCUSSION THREAD

**Discussion by rfl890** [mlua-rs/mlua#396] [Q&A] [answered]
Is there a way to accept userdata struct arguments from Lua without cloning (i.e taking a mutable reference)?

**Comment by khvzak** [comment, marked as answer]
Try `AnyUserData` and then call [`borrow`](https://docs.rs/mlua/latest/mlua/struct.AnyUserData.html#method.borrow) or `borrow_mut` method.

**Comment by rfl890** [comment]
Thanks for the answer!
